### PR TITLE
docs(belt): update Option-module docs

### LINKED ISSUE #3: Transfer Belt.Option docs
We transferred the names and type definitions, but explanations and examples are missing.

Current work: https://github.com/reason-association/reasonml.org/blob/master/pages/belt_docs/option.mdx

Belt Option docs: https://bucklescript.github.io/bucklescript/api/Belt_Option.html

Please comment on this issue if you would like to pick it up.

### PR DESCRIPTION
**Pull request by lessp**
Hey all! Love all your work on this!!

I copied over everything from https://bucklescript.github.io/bucklescript/api/Belt_Option.html, and added a few examples.

Currently, I've added:

```
**Usage**
...
**Examples**
```

To all functions, it'd be nice if these could be anchor-links for easy sharing. But might be something for the future. 🙂

Let me know if there's anything that needs changing!

Closes #3

### CODE REVIEW

**Comment by ryyppy** [comment]
Looking good! 🎉

@lessp I agree with the anchors, we need to refine our md extraction / rendering process a little bit before we can do that, on the other hand, we have anchor links for function names already, so this should be good enough for now :-)

I wonder if *usage* is redundant (since we have examples as well), but let's experiment here!

**Comment by nikgraf** [comment]
This is fantastic! Thanks @lessp

**Comment by baransu** [comment]
@ryyppy I was thinking about **Usage** redundancy and it makes docs really hard to follow.
For me, one of the best examples is Elm package docs. For example: https://package.elm-lang.org/packages/rtfeldman/elm-css/latest/Css#batch where first you have type signature and then detailed explanation with usage examples

**Comment by ryyppy** [comment]
@baransu Yeah, we are currently figuring out the semantic structure of the markdown. There are some limitations we hit with our current setup, since we don't have access to the raw data of the type signatures. We will get there until we found a good way to access odoc data as JSON in a way we need it!
